### DISCUSSION TITLE
Question: Custom circle shape for custom point on Line-Graph.

### DISCUSSION THREAD

**Discussion by parveen-sharma-huru** [patrykandpatrick/vico#386] [Questions]
**Describe the bug**
Hi, first of all thanks for this amazing library. I need to display a custom circle shape on the Line-graph without any touch or scroll behavior. but I'm not able to find the appropriate way to do this. Let' say I have a list of events for the current month and want to display today's event only in the graph, attaching screenshot for the reference.

**Screenshots**
![image](https://github.com/patrykandpatrick/vico/assets/121558741/f8ef9bbe-7b17-4562-bd7e-1e55011e43e0)

**Comment by parveen-sharma-huru** [comment]
@patrickmichalik do you have any solution for this where we need to show only one point from the dataset on the graph?

**Reply by patrickmichalik** [reply]
Hello! You can achieve this by creating a custom persistent `Marker`.

**Reply by parveen-sharma-huru** [reply]
@patrickmichalik thanks for your reply. But how will you achieve the two different text having different styles along with it's dynamic placement around the persistent marker circle (red) position (based on space available) in the graph. Can you please help here?

**Reply by patrickmichalik** [reply]
[`Marker#draw`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.marker/-marker/draw) lets you draw anything you’d like. For the text, you can use [`TextComponent`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.component.text/-text-component). Regarding positioning it, [`TextComponent#getHeight`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.component.text/-text-component/get-height) and [`MeasureContext#canvasBounds`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.context/-measure-context/canvas-bounds) will come in handy. ([`MeasureContext`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.context/-measure-context) is implemented by [`DrawContext`](https://patrykandpatrick.com/vico/api/core/com.patrykandpatrick.vico.core.context/-draw-context).) Data labels have this kind of dynamic positioning, so I suggest that you take a look [here](https://github.com/patrykandpatrick/vico/blob/58d0584a5a6ae61278ef38fd35fbb00dadd9e55d/vico/core/src/main/java/com/patrykandpatrick/vico/core/chart/line/LineChart.kt#L406). More generally, you might want to use [`MarkerComponent#draw`](https://github.com/patrykandpatrick/vico/blob/master/vico/core/src/main/java/com/patrykandpatrick/vico/core/component/marker/MarkerComponent.kt#L75) as a reference.
